### PR TITLE
[WIP] Set up CI workflow

### PR DESCRIPTION
**Pull request by locks**

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -1,17 +1,58 @@
-name: GitHub Actions Demo
-on: [push, pull_request]
+# This workflow uses actions that are not certified by GitHub.  They are
```
Comment (plural) [inline .github/workflows/ci.yml:1]: what's not certified by github here?  This is an area I'm not familiar with, but none of this looks like it's using 3rd party services.  What am I missing>

**Review by plural — COMMENTED** [review]
just a question

File: .github/workflows/ci.yml (hunk 1 above)
Comment (locks) [reply .github/workflows/ci.yml:1]: The GH documentation points to a repo with boilerplate recipes, where I took this from. I think that's what they mean, use this boilerplate at your own peryl :P

**Comment by locks** [comment]
I didn't make it a draft PR or the workflow won't run.

File: .github/workflows/ci.yml (hunk 1 above)
Comment (plural) [reply .github/workflows/ci.yml:1]: hehehe, okee dokee.

**Pull request merged**